### PR TITLE
Fix/man page

### PR DESCRIPTION
**Pull request by anadahz**
This fixes https://trac.torproject.org/projects/tor/ticket/17319

cc: @willscott

### CODE REVIEW

File: data/ooniprobe.1 [hunk 1]
```diff
@@ -119,6 +121,9 @@ Useful when debugging freezes or locks in complex code.
 .TP
 .BR \-\-version
 Display the ooniprobe version and exit.
+.TP
+.BR \-\^Q " or " \-\-queue
+AMQP Queue URL amqp://user:pass@host:port/vhost/queue
```
Comment (willscott) [inline data/ooniprobe.1]: What is an 'AMQP'? This needs a better description.

File: data/ooniprobe.1 [hunk 2]
```diff
@@ -119,6 +121,9 @@ Useful when debugging freezes or locks in complex code.
...
 
 .SH OONIPROBE
 .sp
```
Comment (willscott) [inline data/ooniprobe.1]: on line 130: "researches" -> "researchers"

**Comment by willscott** [comment]
These look good, but there are remaining nits from the ticket in trac that aren't addressed by this, like reference to missing mlab test decks among others. Is this meant to be a combined pull to fix everything?

File: data/ooniprobe.1 (hunk 1 above)
Comment (anadahz) [inline data/ooniprobe.1]: In sort this is used to make ooniprobe work with blocked: https://github.com/openrightsgroup/cmp-issues/issues/78
Not sure how we can describe this in a way that makes senses to the user.

File: data/ooniprobe.1 (hunk 2 above)
Comment (anadahz) [inline data/ooniprobe.1]: Thx fixed.

**Comment by anadahz** [comment]
@willscott Do you think that mlab test decks are relevant to the ooniprobe user?

**Comment by anadahz** [comment]
It will be nice to have an auto generated man page with the actual ooniprobe version and date.

**Comment by willscott** [comment]
I think they aren't relevant, which is why having them referenced in the man page is potentially confusing.

The queue thing needs at least sufficient documentation for someone technical to figure out what's going on. even that github bug leaves it pretty unclear what's going on here. Is this some alternative way to get test input (that's stored in a rabbitmq server somewhere?)

Saying, 'there's this option that causes a set of acronyms to happen' seems like it can only increase confusion.

**Comment by anadahz** [comment]
Removed the mlab deck references.
I agree the AMQP option introduces confusion. 
I'm not aware of a way to get a test input.
Perhaps @dantheta can provide as with some useful test input?

**Comment by dantheta** [comment]
I don't think there is a way to get test input for the AMQP routines - it's intended to point at an AMQP (Rabbit) server.  However, the AMQP options are only useful on the bin/ooniprobequeue command (which uses a variant director), not bin/ooniprobe; I guess the command line options appear for ooniprobe because the command-line options are in shared code.

It might be possible to wrap the parseOptions function so that it uses an extended Options class for the DaemonDirector.  That way the --amqp option wouldn't appear when someone was running bin/ooniprobe --help, only when they were running bin/ooniprobequeue.

**Comment by willscott** [comment]
an initial set of additional changes I'll propose: https://github.com/willscott/ooni-probe/commit/aab516cc433c2cc53bbe0f22d3379f962cc92f69

**Comment by anadahz** [comment]
@willscott apart from the queue comments (https://github.com/willscott/ooni-probe/commit/aab516cc433c2cc53bbe0f22d3379f962cc92f69#commitcomment-18060250) looks good to me.

**Comment by willscott** [comment]
Re-added AMQP help line to my branch https://github.com/TheTorProject/ooni-probe/compare/fix/man-page...willscott:fix/man-page

And filed bug to do it the right way as suggested https://github.com/TheTorProject/ooni-probe/issues/560

Merge/squash my branch into this pull request?

**Comment by hellais** [comment]
@dantheta actually since ooniprobe 1.4.0 (https://github.com/TheTorProject/ooni-probe/commit/1192ca0d0fd7c7b4aa5cb593fe711eb175d5c880 specifically) there is no longer the need to use the specialised ooniprobequeue command to use the messagequeue, but it will invoke that code path upon launching it with the `--queue` option.

You should probably keep this in mind when you update to a later version of ooniprobe for ORG.

**Comment by dantheta** [comment]
Ah, that's really cool.  Thanks!

**Comment by willscott** [comment]
Lets also try to improve the documentation of the -p option, which is unclear (https://trac.torproject.org/projects/tor/ticket/11988)

**Comment by hellais** [comment]
I believe this is now superseeded by the manpage docs being generated from markdown files. I believe all the feedback in here should be integrated in there, but please double check to be sure.

**Comment by hellais** [comment]
Can I close this?

**Comment by anadahz** [comment]
There is significant amount of documentation missing from the markdown docs that should either be added to the docs itself or referenced to the manpages as links or external files (eg: in /usr/share/doc).

The documentation topics that are not covered by the current [markdown manual pages](https://github.com/TheTorProject/ooni-probe/tree/master/docs/source/manual) are:
- Default collector, bouncer addresses
- Test enumeration and description
- Default paths for reports, decks and config files
- OONI terms explanation (tests, decks, test helper,...)
- Configuring ooniprobe
- Running decks
- Running net tests
- Optional packages (obfs4proxy)
- Bridges and obfsproxy bridges
- Third party tests and dependencies

**Comment by hellais** [comment]
> There is significant amount of documentation missing from the markdown docs that should either be added to the docs itself or referenced to the manpages as links or external files (eg: in /usr/share/doc).

Sounds good. Can you add these to the markdown files as part of a new PR? I also don't think all of these need to necessarily be present inside of the manpage as they are documented elsewhere.

**Comment by anadahz** [comment]
@hellais Sure, though if this issue is closed it will be most probably forgotten.

> I also don't think all of these need to necessarily be present inside of the manpage as they are documented elsewhere.

Which places are you referring to?
It makes sense to add a URL in the man pages pointing to the documentation already mentioned elsewhere.

**Comment by hellais** [comment]
> Which places are you referring to?

The README has most of these things.

> It makes sense to add a URL in the man pages pointing to the documentation already mentioned elsewhere.

Sure go for it!

**Pull request closed without merging**